cmd/cdpgen/main.go

```diff
@@ -814,19 +814,24 @@ func (g *Generator) domainTypeStruct(d proto.Domain, t proto.AnyType) {
 }
 
 func (g *Generator) domainTypeTime(d proto.Domain, t proto.AnyType) {
+	var mp int
+	if d.Name() == "Runtime" {
+		// Runtime domain denotes timestamps in milliseconds.
+		mp = 1000000
+	} else {
+		mp = 1000000000
+	}
+
 	g.Printf(`float64
 
 // String calls (time.Time).String().
 func (t %[1]s) String() string {
 	return t.Time().String()
 }
 
-// Time parses the Unix time with millisecond accuracy.
+// Time parses the Unix time.
 func (t %[1]s) Time() time.Time {
-	secs := int64(t)
-	// The Unix time in t only has ms accuracy.
-	ms := int64((float64(t)-float64(secs))*1000000)
-	return time.Unix(secs, ms*1000)
+	return time.Unix(0, int64(float64(t)*%[3]d))
```

Comment:
This is a neat solution to the problem, but I'm a bit vary of the potential for integer overflow, which is why it was written differently previously. I realize this won't realistically happen until somewhere around the year 2300, but it could happen if the time of a machine was severely skewed or performing some weird future testing.

What do you think, needless worry?

Reply:
I thought we don't need to worry about Y2300, but given that the original code could handle it, I agree we don't need to introduce regression.

I've updated the code. Thanks for pointing this out!